src/view.js

```diff
@@ -180,64 +103,11 @@ Marionette.View = Marionette.AbstractView.extend({
     if (this.getOption('destroyImmediate') === true) {
       this.$el.remove();
     }
-    this.regionManager.destroy();
-    return Marionette.AbstractView.prototype.destroy.apply(this, arguments);
-  },
 
-  // Internal method to initialize the regions that have been defined in a
-  // `regions` attribute on this layoutView.
-  _initializeRegions: function(options) {
-    this._initRegionManager();
+    this.removeRegions();
 
-    this._addRegions(this.regions, [options]);
-    this._addRegions(this.getOption.call(options, 'regions'), [options]);
-  },
-
-  // internal method to build regions
-  _buildRegions: function(regions) {
-    var defaults = {
-      regionClass: this.getOption('regionClass'),
-      parentEl: _.partial(_.result, this, 'el')
-    };
-
-    return this.regionManager.addRegions(regions, defaults);
-  },
-
-  // Internal method to re-initialize all of the regions by updating
-  // the `el` that they point to
-  _reInitializeRegions: function() {
-    this.regionManager.invoke('reset');
-  },
-
-  // Internal method to initialize the region manager
-  // and all regions in it
-  _initRegionManager: function() {
-    this.regionManager = this.getRegionManager();
-    this.regionManager._parent = this;
-
-    this.listenTo(this.regionManager, 'before:add:region', function(name) {
-      this.triggerMethod('before:add:region', name);
-    });
-
-    this.listenTo(this.regionManager, 'add:region', function(name, region) {
-      this[name] = region;
-      this.triggerMethod('add:region', name, region);
-    });
-
-    this.listenTo(this.regionManager, 'before:remove:region', function(name) {
-      this.triggerMethod('before:remove:region', name);
-    });
-
-    this.listenTo(this.regionManager, 'remove:region', function(name, region) {
-      delete this[name];
-      this.triggerMethod('remove:region', name, region);
-    });
-  },
-
-  _getImmediateChildren: function() {
-    return _.chain(this.regionManager.getRegions())
-      .pluck('currentView')
-      .compact()
-      .value();
+    return Marionette.AbstractView.prototype.destroy.apply(this, arguments);
   }
 });
+
+_.extend(Marionette.View.prototype, Marionette.RegionsMixin);
```

Comment:
nope, i'm going to remove metal too. It's currently flagged off, but still a headache